exchanges/okcoin/okcoin.go

```
@@ -1,22 +1,749 @@
+	requestURL := orders + formatParameters(request)
+	var resp []GetSpotOrderResponse
+	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, marginTradingSubsection, requestURL, nil, &resp, true)
+}
+
+// GetMarginOpenOrders List all your current open orders. Cursor pagination is used. All paginated requests return the latest information (newest) as the first page sorted by newest (in chronological time) first.
+func (o *OKCoin) GetMarginOpenOrders(ctx context.Context, request *GetSpotOpenOrdersRequest) ([]GetSpotOrderResponse, error) {
+	requestURL := pendingOrders + formatParameters(request)
+	var resp []GetSpotOrderResponse
+	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, marginTradingSubsection, requestURL, nil, &resp, true)
+}
+
+// GetMarginOrder Get order details by order ID.
+func (o *OKCoin) GetMarginOrder(ctx context.Context, request *GetSpotOrderRequest) (*GetSpotOrderResponse, error) {
+	requestURL := orders + "/" + request.OrderID + formatParameters(request)
+	var resp *GetSpotOrderResponse
+	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, marginTradingSubsection, requestURL, request, &resp, true)
+}
+
+// GetMarginTransactionDetails Get details of the recent filled orders. Cursor pagination is used.
+// All paginated requests return the latest information (newest) as the first page sorted by newest (in chronological time) first.
+func (o *OKCoin) GetMarginTransactionDetails(ctx context.Context, request *GetSpotTransactionDetailsRequest) ([]GetSpotTransactionDetailsResponse, error) {
+	requestURL := getSpotTransactionDetails + formatParameters(request)
+	var resp []GetSpotTransactionDetailsResponse
+	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, marginTradingSubsection, requestURL, nil, &resp, true)
+}
+
+// formatParameters Formats URL parameters, useful for optional parameters due to OKCoin signature check
+func formatParameters(request interface{}) string {
```

Suggestion: I would return an error.